05_iigo/index.tex

```diff
@@ -551,8 +546,78 @@ \subsubsection{Server-side}
+\subsection{Legislative Branch}
+\subsection{Judicial Branch}
+The judicial branch has two parts: the Judicial branch itself, which exists on the server to perform the mechanical actions required for IIGO, and the judge role which is in the hands of the agent which is currently Judge. This agent is able to use the Judge role to make decisions which  are communicated to the serverside Judicial-branch which performs the actions.
+\subsubsection{Client-side}
+The Judge decision making functions can be divided into two categories:
+\begin{enumerate}
+    \item Judge power functions:
+    \begin{itemize}
+        \item \texttt{InspectHistory} allows the Judge to inspect the history cache and evaluate whether agents have complied with the rules or transgressed. Provided the history cache, this returns a $map[agent] \xrightarrow[]{} []transgressions$ as well as information about whether the Judge has acted on its power to inspect the history cache.
+        \item \texttt{GetRuleViolationSeverity} allows the Judge to set sanction penalties for violating specific rules. Returns a $map[ruleName] \xrightarrow[]{} penalty$. If this is not set by the Judge, the default sanction penalty is 1 unit of resources for every rule broken.
+        \item \texttt{GetSanctionThresholds} allows the Judge to set the sanction penalty thresholds for sanction tiers. Returns a $map[sanctionTier] \xrightarrow[]{} penalty$.
+        \item \texttt{GetPardonedIslands} allows the Judge to pardon islands of imposed sanctions. Provided with the list of sanctions, this returns a $map[sanctionID] \xrightarrow[]{} bool$ where a $true$ value indicates that the sanction corresponding to that $ID$ is to be removed from the list of sanctions.
+        \item \texttt{HistoricalRetributionEnabled} allows the Judge to decide whether to inspect the history and accordingly sanction unpunished transgressions from previous turns. This is a boolean value so that historical retribution can be enabled and disabled easily.
+    \end{itemize}
+    \item Transfer-of-power functions:
+    \begin{itemize}
+        \item \texttt{PayPresident} implements the obligation of the Judge to pay the President its salary. 
+        \item \texttt{CallPresidentElection} implements the power of the Judge to initiate the transfer-of-power for the President.
+        \item \texttt{DecideNextPresident} allows the Judge to modify the outcome of the election and choose not to follow its obligation to announce the result of President election.
+    \end{itemize}
+\end{enumerate}
+\subsubsection{Server-side}
+The functionality of the server-side of the Judicial branch can be summarised as follows:
+\begin{enumerate}
+    \item Information about actions taken by Islands is used to populate the history cache which is an array of $map[agent] \xrightarrow[]{} action$. This is done server-side because the history cache is considered to be an absolute source of truth i.e. islands do not have the opportunity to lie about actions.
```

Comment:
```suggestion
    \item Information about actions taken by Islands is used to populate the history cache which is an array of $map[agent] \xrightarrow[]{} action$. This is done server-side because the history cache is considered to be an absolute source of truth, i.e. islands do not have the opportunity to lie about actions.
```